packages/eslint-plugin/src/rules/switch-exhaustiveness-check.ts

```diff
@@ -128,10 +128,21 @@ export default createRule({
 
           caseTypes.add(getNodeType(switchCase.test));
         }
+        const missingBranchTypes = unionTypes.filter(unionType => {
+          const type = isTypeFlagSet(
+            unionType,
+            ts.TypeFlags.Number |
+              ts.TypeFlags.String |
+              ts.TypeFlags.Object |
+              ts.TypeFlags.Intersection,
+          );
+
+          if (type) {
+            return caseTypes.has(unionType);
+          }
```

Comment:
This is incorrect—if a union constituent is infinite, you should just bail. Consider this case:

```js
declare const a: number | string;
declare const b: number;
switch (a) {
  case b: break;
}
```

It still reports "not exhaustive" and results in an invalid fix.